Upgrade from Haskell2010 to GHC2021

locallycompact opened the pull request:
This moves the default-language to GHC2021 and reduces the extension list in all cabal files.

locallycompact committed:
Upgrade from Haskell2010 to GHC2021

ch1bo commented on hydra-cardano-api/hydra-cardano-api.cabal:
```diff
@@ -19,45 +19,19 @@ flag hydra-development
 common project-config
-  default-language:   Haskell2010
+  default-language:   GHC2021
   default-extensions:
```
@locallycompact fixed the `plutus-...` packages as well

ch1bo reviewed (approved):
Checked and all GHC2021 extensions we had before or are inoffensive.

I also pushed a commit that removes several in-module LANGUAGE pragmas as well @locallycompact

ch1bo committed:
Remove redundant inline LANGUAGE pragmas

This also moved PatternSynonyms into the .cabal file in packages which
use it often.
